DashSync/Models/DSPeer.m

```diff
@@ -186,30 +185,31 @@ - (void)connect
     if (self.status != DSPeerStatus_Disconnected) return;
     _status = DSPeerStatus_Connecting;
     _pingTime = DBL_MAX;
-    if (! self.reachability) self.reachability = [Reachability reachabilityForInternetConnection];
+    if (! self.reachability) self.reachability = [DSReachabilityManager sharedManager];
     
-    if (self.reachability.currentReachabilityStatus == NotReachable) { // delay connect until network is reachable
+    if (self.reachability.networkReachabilityStatus == DSReachabilityStatusNotReachable) { // delay connect until network is reachable
         NSLog(@"%@:%u not reachable, waiting...", self.host, self.port);
         
         dispatch_async(dispatch_get_main_queue(), ^{
             if (! self.reachabilityObserver) {
                 self.reachabilityObserver =
-                [[NSNotificationCenter defaultCenter] addObserverForName:kReachabilityChangedNotification object:nil
+                [[NSNotificationCenter defaultCenter] addObserverForName:DSReachabilityDidChangeNotification object:nil
                                                                    queue:nil usingBlock:^(NSNotification *note) {
-                                                                       if (self.reachabilityObserver && self.reachability.currentReachabilityStatus != NotReachable) {
+                                                                       if (self.reachabilityObserver && self.reachability.networkReachabilityStatus != DSReachabilityStatusNotReachable) {
                                                                            self->_status = DSPeerStatus_Disconnected;
                                                                            [self connect];
                                                                        }
                                                                    }];
                 
-                [self.reachability startNotifier];
+                if (!self.reachability.monitoring) {
+                    [self.reachability startMonitoring];
+                }
             }
         });
         
         return;
     }
     else if (self.reachabilityObserver) {
-        [self.reachability stopNotifier];
```

Comment:
Is this no longer needed? Basically to stop monitoring.

Reply:
Same for a few lines down.

Reply:
Yep, because we constantly monitoring reachability. There is no much sense of turning it on/off every 60 seconds (our ticker refresh interval) as it may also lead to delays before operation start